rust/src/transformations/make_stable_lazyframe/truncate/matching/test.rs

```diff
@@ -0,0 +1,298 @@
+use polars::{
+    df,
+    prelude::{IntoLazy, RankOptions, all_horizontal, as_struct, col},
+};
+
+use super::*;
+
+#[test]
+fn test_match_truncations_ok() -> Fallible<()> {
+    let lf = df!["A" => [1], "B" => [1], "ID" => [1]]?.lazy();
+
+    let rank_opt = RankOptions {
+        method: RankMethod::Dense,
+        descending: false,
+    };
+    let truncate_num_groups = col("A").rank(rank_opt, None).over([col("ID")]).lt(lit(10));
+
+    let plan = lf
+        .clone()
+        .filter(truncate_num_groups.clone())
+        .group_by([col("ID"), col("A")])
+        .agg([col("B").sum()]);
+
+    let (_, truncations, truncation_bounds) =
+        match_truncations(plan.logical_plan.clone(), &col("ID")).unwrap();
+    assert_eq!(
+        truncations,
+        vec![
+            Truncation::Filter(truncate_num_groups),
+            Truncation::GroupBy {
+                keys: vec![col("ID"), col("A")],
+                aggs: vec![col("B").sum()]
+            },
+        ]
+    );
+    assert_eq!(
+        truncation_bounds,
+        vec![
+            Bound {
+                by: HashSet::from([col("A")]),
+                per_group: None,
+                num_groups: Some(10),
+            },
+            Bound {
+                by: HashSet::from([col("A")]),
+                per_group: Some(1),
+                num_groups: None,
+            },
+        ]
+    );
+
+    Ok(())
+}
+
+#[test]
+fn test_match_truncations_groupby_not_last() -> Fallible<()> {
+    let lf = df!["A" => [1], "B" => [1], "ID" => [1]]?.lazy();
+
+    let rank_opt = RankOptions {
+        method: RankMethod::Dense,
+        descending: false,
+    };
+    let truncate_num_groups = col("A").rank(rank_opt, None).over([col("ID")]).lt(lit(10));
+
+    let plan = lf
+        .group_by([col("ID"), col("A")])
+        .agg([col("B").sum()])
+        .filter(truncate_num_groups.clone());
+
+    // should get a nice error when group by truncation is not last
+    assert!(match_truncations(plan.logical_plan.clone(), &col("ID")).is_err());
```

Comment:
I'm guessing there's another place I commented about testing an error message that might be caught here. Maybe check the error message? (Kind of a stop gap for measuring test coverage.)